### PR TITLE
Add a trivial conformance test.

### PR DESCRIPTION
**Pull request by toddjonker**
*Description of changes:*
Adding a simple test to enable initial implementation and integration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zslayton — APPROVED** [review]
We should take this opportunity to add comments to all of the tests (as we write them) indicating what the test is aiming to verify. Sometimes it's obvious, but often it's not.

**Comment by toddjonker** [comment]
> We should take this opportunity to add comments to all of the tests (as we write them) indicating what the test is aiming to verify. Sometimes it's obvious, but often it's not.

I agree, and I think this is an obvious case, no?

BTW I also expect that the suite and the DSL will be evolved regularly as we learn more about what we need.  This will be very much a living product.

**Comment by zslayton** [comment]
> I agree, and I think this is an obvious case, no?

It's obvious to Ion maintainers, yes. I don't think it would be obvious to someone implementing Ion for the first time.

**Comment by toddjonker** [comment]
> It's obvious to Ion maintainers, yes. I don't think it would be obvious to someone implementing Ion for the first time.

I think individual test files should come with a baseline assumption that the reader knows the testing DSL.  We shouldn't have to document fundamentals all over the place.  And once you take that out of the picture, if you assume the reader understands how the testing DSL works, what remains here is trivial.

Or maybe I'll put it this way: if these three LoC are not obvious to the Ion implementer, we should respond by improving the README documentation so they can better understand how the tests work.

**Comment by zslayton** [comment]
I'm not referring to the DSL framework, I mean the intent of the test itself. We're demonstrating equality between `null` and `null.null`, while also showing that an annotated `null` is equivalent to a `null` with the same annotations.

**Comment by toddjonker** [comment]
Oh I see, the test is _too_ trivial.  You're right, at this point its exercising the test framework more than anything.  Would make more sense using `text` and/or `binary` but I don't have those implemented.

File: conformance/null.ion [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2023-2024 Amazon.com, Inc.  All rights reserved.
```
Comment (popematt) [inline conformance/null.ion:1]: Can we leave out the dates from the copyright header? We've been using this header for `ion-java` and `ion-rust`, which both also use the Apache-2.0 license.
```suggestion
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
```

**Review by tgregg — APPROVED** [review]

File: conformance/null.ion (hunk 1 above)
Comment (toddjonker) [reply conformance/null.ion]: Thanks, I didn't realize the recommendations had changed.

**Pull request merged**